tests/unit/types/document/test_document.py

```diff
@@ -828,3 +732,105 @@ def test_doc_match_score_assign():
     d1 = Document(d, copy=True, score=123)
     d.matches = [d1]
     assert d.matches[0].score.value == 123
+
+
+def test_doc_update_given_empty_fields_and_attributes_identical(test_docs):
+    # doc1 and doc2 has the same fields, id, content, tags, embedding and chunks.
+    doc1, doc2 = test_docs
+    doc1.update(source=doc2)
+    assert doc1.id == doc2.id
+    assert doc1.content == doc2.content
+    assert doc1.tags == doc2.tags
+    assert (doc1.embedding == doc2.embedding).all()
+    assert doc1.chunks == doc2.chunks
+
+
+def test_doc_update_given_empty_fields_and_destination_has_more_attributes(test_docs):
+    # doc1 and doc2 has the same fields, id, content, tags, embedding and chunks.
+    doc1, doc2 = test_docs
+    # remove doc2 content field
+    doc2._pb_body.ClearField(
+        'content'
+    )  # content of source "goodbye-world" was removed, not update this field.
+    assert doc2.content is None
+    doc1.update(source=doc2)
+    assert doc1.id == doc2.id
+    assert doc1.content == 'hello-world'  # doc1 content remains the same.
+    assert doc1.tags == doc2.tags
+    assert (doc1.embedding == doc2.embedding).all()
+    assert doc1.chunks == doc2.chunks
+
+
+def test_doc_update_given_empty_fields_and_source_has_more_attributes(test_docs):
+    # doc1 and doc2 has the same fields, id, content, tags, embedding and chunks.
+    doc1, doc2 = test_docs
+    # remove doc2 content field
+    doc1._pb_body.ClearField('content')  # content of destination was removed.
+    assert doc1.content is None
+    doc1.update(source=doc2)
+    assert doc1.id == doc2.id
+    assert (
+        doc1.content == doc2.content
+    )  # destination content `None` was updated by source's content.
+    assert doc1.tags == doc2.tags
+    assert (doc1.embedding == doc2.embedding).all()
+    assert doc1.chunks == doc2.chunks
+
+
+def test_doc_update_given_singular_fields_and_attributes_identical(test_docs):
+    # doc1 and doc2 has the same fields, id, content, tags, embedding and chunks.
+    doc1, doc2 = test_docs
+    # After update, only specified fields are updated.
+    doc1.update(source=doc2, fields=['id', 'content'])
+    assert doc1.id == doc2.id
+    assert doc1.content == doc2.content  # None was updated by source's content.
+    assert doc1.tags != doc2.tags
+    assert (doc1.embedding != doc2.embedding).all()
+    assert doc1.chunks != doc2.chunks
+
+
+def test_doc_update_given_nested_fields_and_attributes_identical(test_docs):
+    # doc1 and doc2 has the same fields, id, content, tags, embedding and chunks.
+    doc1, doc2 = test_docs
+    # After update, only specified nested fields are updated.
+    doc1.update(source=doc2, fields=['tags', 'embedding', 'chunks'])
+    assert doc1.id != doc2.id
+    assert doc1.content != doc2.content  # None was updated by source's content.
+    assert doc1.tags == doc2.tags
```

Comment:
As for the `tags`, it is better to stay consistent with the [python `update` method](https://docs.python.org/3/library/stdtypes.html?highlight=update#dict.update). To be specific, if the key exists, replace the old value with the new one. Otherwise, append the key-value. 

```suggestion
    target = {'a': 'b'}
    target.update({'c': 'd'})
    assert doc1.tags == target
```

Reply:
@nan-wang code updated

Reply:
> As for the tags, it is better to stay consistent with the python update method.
im not really convinced by this logic, see my comment below